### DISCUSSION TITLE
horizontal scroll

### DISCUSSION THREAD

**Discussion by malordin** [sqlitebrowser/sqlitebrowser#3497] [Q&A] [answered]
Hey, everybody!

I have a small database that I open in SQLitebrowser for normal browsing as no other functionality is required.

Everything works perfectly, but I could use a horizontal scroll function because some lines go beyond my screen. In order to scroll, you have to move the mouse cursor to the bottom scroll bar, which is not very convenient. I would like to do it with Shift + Scroll UP\Scroll Down on the mouse, as it is done in any other popular applications.

I looked at the settings in the program and found nothing. Also looked for information in issues and here in discussions and found nothing either.

Maybe I am missing something? I would welcome any response, thanks in advance.

**Comment by lucydodo** [comment]
Hi @malordin 
When testing on my system*, I confirmed that horizontal scrolling is possible with Shift + Scroll Up/Down.

For which parts of the UI (e.g. 'Browse Data' or 'Edit Database Cell', etc.) have you by any chance tried this?
>*: macOS 14.0 (23A344) and Logitech MX Anywhere 3

**Reply by malordin** [reply]
Thanks for the reply, my system is Win 10 21H2 , but Shift+Scroll Up/Down is not working. I have recorded a gif for an illustrative example.

Instead of doing horizontal scroll the page does scroll down or up only in larger increments of 50 values.

![sql](https://github.com/sqlitebrowser/sqlitebrowser/assets/20083687/4bfb9cdb-eb90-4c3f-b741-f6a02e2ff285)

**Reply by lucydodo** [reply]
That's weird, it works fine too when I test it on my Windows 11 Version 22H2.
Do you have any software installed that does some sort of remapping for the mouse?

> Note: I used the Belfast Bikes Docking Stations database file for tests.
> You can download the db file via [here](https://nightlies.sqlitebrowser.org/example_datasets/Belfast%20Bikes%20Docking%20Stations/)

**Reply by malordin** [reply]
I downloaded the database you specified, nothing changed. But I just discovered that horizontal scrolling works via alt+Scroll Up\Down. It's clearer, but is there any way to swap the functionality? So that horizontal scrolling would be on shift, and vertical scrolling with large segments on alt?

**Reply by lucydodo** [reply]
Thankfully, I learned something new.
When I pressed 'Alt' instead of 'Shift' and tried to scroll, I noticed that the amount of scrolling was different.

Presumably the default is 'Alt', and this is just a guess on my part,
but I think the Windows API provides for 'Shift' to be used for horizontal scrolling
> At least the macOS Cocoa API is designed that way, but I don't know much about the Windows API,
> so this is just a my guess.

Anyway, I think the bottom line is that it's a bug. I'm afraid I can't fix this right now because I have other things to do, so it may be inconvenient until this issue is resolved, but please use 'Alt' to scroll. and thanks for reporting this. 😄

**Reply by lucydodo** [reply, marked as answer]
I've opened a ticket to discuss and track this bug: https://github.com/sqlitebrowser/sqlitebrowser/issues/3498
